src/librustc_codegen_ironox/ir/instruction.rs

```diff
@@ -0,0 +1,42 @@
+use context::CodegenCx;
+use ir::type_::Type;
+use ir::value::Value;
+use super::super::ModuleIronOx;
+
+use rustc::ty::layout::Align;
+use rustc_codegen_ssa::traits::BaseTypeMethods;
+
+/// An IronOx instruction.
+#[derive(PartialEq, Debug, Eq, Hash)]
+pub enum Instruction {
+    /// Call(fn_idx, args). Emit a call to the function found at index `fn_idx`
+    /// in the `functions` vector of the module.
+    Call(usize, Vec<Value>),
+    /// Allocate space on the stack for a variable of a particular type and
+    /// alignment.
+    Alloca(String, Type, Align),
+    /// Cast a value to a type.
+    Cast(Value, Type),
+    /// Add two values and return the result.
+    Add(Value, Value),
+    /// Currently serves as a no-op. This variant will be removed.
```

Comment:
I don't think there is any reason why the IR should contain no-ops (at least not at the moment). I was using it because I wanted to compile the backend without emitting certain instructions (i was going to replace all `Instruction::None`s with real instructions). I'll remove it, since it doesn't have any real purpose here.

Reply:
OK.